frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { AuthService } from '@core/services/auth.service';
+import { PasswordsErrorStateMatcher } from '@modules/auth/validators/passwordsErrorStateMatcher';
+
+@Component({
+    selector: 'app-sign-up-form',
+    templateUrl: './sign-up-form.component.html',
+    styleUrls: ['./sign-up-form.component.sass', '../../shared-styles.sass'],
+})
+export class SignUpFormComponent {
+    public matcher = new PasswordsErrorStateMatcher();
+
+    public hidePassword = true;
+
+    public signUpForm = new FormGroup({
+        email: new FormControl('', { validators: [Validators.required, Validators.email], updateOn: 'submit' }),
+        name: new FormControl('', {
+            validators: [
+                Validators.required,
+                Validators.minLength(3),
+                Validators.maxLength(50),
+                Validators.pattern('^[a-zA-Zа-яА-Я0-9- ]+$'),
```

Comment:
Could you please create a separate file (if not exist) and move this regex into this file?